lib/roger/resolver.rb

```diff
@@ -176,7 +176,7 @@ def sanitize_name(name, prefer = nil)
       path = name.to_s
 
       # If it's a directory append "index"
-      path = File.join(path, "index") if File.directory?(name)
+      path = File.join(path, "index") if File.directory?(name) && Dir.glob(name + ".*").none?
```

Comment:
This is potentially a faulty check as you are essentially looking for "bla.*" so bla with an extension when in fact you want to check for "bla/.*", correct?